### PR TITLE
#11132 Run Post-Commit Python Tests against wheels: Try #2

### PR DESCRIPTION
**Pull request by dimitri-tenstorrent**
The purpose of this PR is to introduce running of Post Commit Python tests in the isolated Docker environment with the Python environment setup by installing Wheels.

### Ticket
https://github.com/tenstorrent/tt-metal/issues/8082
https://github.com/tenstorrent/tt-metal/issues/11132
https://github.com/tenstorrent/tt-metal/issues/11257

### Problem description
We need to make sure the wheels that we publicize to our users have been tested.
We need to test the Release version of Docker image with wheels installed ( this will need the definition of the release image)
Also, we need to be moving towards dockerization of CI/CD and this is a small step towards that.

### What's changed
The order of jobs in All Post Commit Workflow was re-arranged to set build-wheels job as a prerequisite of jobs containing Python tests. The wheels that are built are installed into the Docker environment and these resulting Docker images are pushed to Docker registry.
The Python Tests fast-dispatch-unit-tests, ttnn-unit-tests and model-unit-tests are now executed inside of the Docker container with wheel.

### Checklist
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: .github/workflows/build-docker-artifact.yaml [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
-
+  push:
+    branches:
+      - main
+    paths:
+      - dockerfile/ubuntu-20.04-amd64.Dockerfile
+      - scripts/docker/requirements*
+      - pyproject.toml
+      - scripts/docker/install_test_deps.sh
+      - tt_metal/python_env/requirements-dev.txt
```
Comment (tt-rkim) [inline .github/workflows/build-docker-artifact.yaml:13]: Mind opening an issue to add requirements-infra to this list and assigning to me? Don't wanna block

File: .github/workflows/build-docker-artifact.yaml [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
     env:
       TT_METAL_ENV: ${{ vars.TT_METAL_ENV }}
       CONFIG: ci
       SILENT: 0
       VERBOSE: 1
-      TT_METAL_DOCKER_IMAGE: tt-metalium/ubuntu-20.04-amd64
-      TT_METAL_DOCKER_IMAGE_TAG: ${{ github.sha }}
+      IMAGE: tt-metalium/ubuntu-20.04-amd64
+      DOCKERFILE: ubuntu-20.04-amd64
+      IMAGE_TAG: ${{ github.sha }}
```
Comment (tt-rkim) [inline .github/workflows/build-docker-artifact.yaml:23]: Can we delete this if it gets overridden in all cases later?

File: .github/workflows/ttnn-post-commit.yaml [hunk 3]
```diff
@@ -73,23 +72,24 @@ jobs:
...
-          cd $TT_METAL_HOME
-          export PYTHONPATH=$TT_METAL_HOME
-          ${{ matrix.test-group.cmd }}
+        uses: ./.github/actions/docker-run
+        with:
+          docker_username: ${{ github.actor }}
+          docker_password: ${{ secrets.GITHUB_TOKEN }}
+          docker_image_arch: ${{ inputs.arch }}
+          run_args: |
+            WHEEL_FILENAME=$(ls -1 *.whl)
```
Comment (tt-rkim) [inline .github/workflows/ttnn-post-commit.yaml:90]: I'm wondering if we should open an issue to eventually combine prepare metal run, run docker action, and this into something later.

File: scripts/docker/requirements-20.04.txt [hunk 4]
```diff
@@ -19,3 +19,5 @@ python3-dev=3.8.2-0ubuntu2
 python3.8-venv
 cargo
 ninja-build
+patchelf
+graphviz
```
Comment (tt-rkim) [inline scripts/docker/requirements-20.04.txt:23]: POSIX new line

**Review by tt-rkim — APPROVED** [review]

File: .github/workflows/build-docker-artifact.yaml (hunk 2 above)
Comment (ttmchiou) [inline .github/workflows/build-docker-artifact.yaml:23]: we don't need IMAGE_TAG anymore here with github SHA do we?

File: scripts/docker/requirements-20.04.txt (hunk 4 above)
Comment (dimitri-tenstorrent) [reply scripts/docker/requirements-20.04.txt:23]: I'd love to build a static check for this :D

File: .github/workflows/build-docker-artifact.yaml (hunk 1 above)
Comment (dimitri-tenstorrent) [reply .github/workflows/build-docker-artifact.yaml:13]: https://github.com/tenstorrent/tt-metal/issues/12233

File: .github/workflows/ttnn-post-commit.yaml (hunk 3 above)
Comment (dimitri-tenstorrent) [reply .github/workflows/ttnn-post-commit.yaml:90]: There is a very natural next task to create release images that ties all of this together. Not sure if that's what you were thinking about or something else.

**Pull request merged**